### ISSUE TITLE
Expose the insight component as a JavaScript API in the NPM package

### ISSUE THREAD

**Issue by AttilaMihaly** [finos/morphir-elm#268]
Task list:
- [ ] Create an Elm application called `Insight.elm` inside the `cli` folder.
  - The application should be initialized similar to this: https://github.com/finos/morphir-elm/blob/6be48f261a69292395b57619926d3ec973f4664f/cli/src/Morphir/Web/DevelopMain.elm#L29-L35
  - It should take the IR through Flags as described here: https://guide.elm-lang.org/interop/flags.html
    - The output of `morphir-elm make` is a `Distribution` that can be decoded from JSON using this function: https://github.com/finos/morphir-elm/blob/6be48f261a69292395b57619926d3ec973f4664f/src/Morphir/IR/Distribution/Codec.elm#L65-L66
  - It should take the value expression through a Port as described here: https://guide.elm-lang.org/interop/ports.html
    - The value expression will be passed in as a `Value` that can be decoded from JSON using this function: https://github.com/finos/morphir-elm/blob/6be48f261a69292395b57619926d3ec973f4664f/src/Morphir/IR/Value/Codec.elm#L200-L201
    - The decode function requires two additional decoders for type and value annotations. We don't expect any annotations to be passed in so we can pass `Decode.succeed ()` to always pass Unit without decoding.
  - Once both of those are available the `view` function should invoke the visualization code the same way as it's done here: https://github.com/finos/morphir-elm/blob/6be48f261a69292395b57619926d3ec973f4664f/cli/src/Morphir/Web/DevelopMain.elm#L239-L241
- [ ] Add a build step to compile `Insight.elm` into `insight.js`.
- [ ] Add `insight.js` to the `files` section in `package.json`
- [ ] Publish the NPM package as described here: https://github.com/finos/morphir-elm/blob/master/docs/publishing.md
  - This step can only be done by a maintainer for now.

**Comment by AttilaMihaly** [comment]
Building the Elm application will require understanding The Elm Architecture (TEA) so if you are not familiar please read this first: https://guide.elm-lang.org/architecture/

Let me add more details on the design. The component's lifecycle should be the following:

- To initialize the component the consumer should pass in the JSON that was previously generated by `morphir-elm make`. This should be done by using Flags.
- The Elm application should decode the JSON into a `Morphir.IR.Distribution.Distribution` using the decoder mentioned above.
- If the decoding succeeds the application should store the `Distribution` in the model and show an empty <div> to signl successful initialization. 
- It the decoding fails it should show an error message.
- When the consumer is ready to specify which function should be explained they should pass in the fully-qualified name of the function. This should be done using a Port.
- Based on the fully-qualified name the application should look up the function definition in the `Distribution`.
- If the function is found it should invoke `Morphir.Visual.ViewValue.view` with the body of the function. This will show a visualization of the function **without data**.
- If the function is not found it should show an error message.
- When the consumer is ready to pass in the arguments for the function they should pass in a list of arguments as a list of JSON objects that can be decoded into `Morphir.IR.Value.Value` as described above. This should be done through a port.
- The application should decode each argument and pass it to the `Morphir.Visual.ViewValue.viewWithData` function. If the function doesn't exist please use a placeholder that simply ignores the data and calls the `view` function instead.
- If the decoding fail the app should display an error message.

To sum up the application will interact with JavaScript through:
- Flags to get the `Distribution` at startup.
- A port to get the fully-qualified name of the function to explain (without data).
- A port to get  the arguments of the function to explain with data.

The model should capture the following states of the application:
- Initialized with a distribution - this should show an empty <div>.
- Function name set - this should show the explanation without data.
- Function arguments set - this should show the explanation with data.
- Failed - this should show an error message.

**Comment by piyush-gupta1999** [comment]
Is there any function defined in the Distribution file which can lookup function definition in distribution?

**Comment by AttilaMihaly** [comment]
> Is there any function defined in the Distribution file which can lookup function definition in distribution?

No. There is a way to look up a specification but not a definition yet. Please implement it using the `lookupValueSpecification` as a template.

**Comment by AttilaMihaly** [comment]
Merged a PR that added some of the missing features mentioned above: https://github.com/finos/morphir-elm/pull/278
